src/MiniCover/Reports/OpenCoverReport.cs

```diff
@@ -0,0 +1,147 @@
+                    var hitInstructions = file.Value.Instructions.Where(h => hits.Contains(h.Id)).ToArray();
+
+                    return file.Value.Instructions
+                        .GroupBy(instruction => new { instruction.Class })
+                        .Select(classes =>
+                    {
+                        var classElement = new XElement(
+                            XName.Get("Class")
+                        );
+
+                        var classfullNameElement = new XElement(
+                            XName.Get("FullName"),
+                            new XText(classes.Key.Class)
+                        );
+
+                        var methodsList = classes
+                            .GroupBy(instruction => new { instruction.Method, instruction.MethodFullName })
+                            .Select(method => 
+                            {
+                                var nameElement = new XElement(
+                                    XName.Get("Name"),
+                                    new XText(method.Key.MethodFullName)
+                                );
+
+                                var fileRefElement = new XElement(
+                                    XName.Get("FileRef"),
+                                    new XAttribute(XName.Get("uid"), dctSourceFileCount[file.Value])
+                                );
+
+                                int sequencePointMiniCounter = 0;
+                                bool allLinesAreVisited = true;
+
+                                var sequencePoints = method.Select(methodPoint =>
+                                {
+                                    var hitCount = hitInstructions.Count(hit => hit.Equals(methodPoint));
+
+                                    if (hitCount == 0)
+                                        allLinesAreVisited = false;
+
+                                    return new XElement(
+                                        XName.Get("SequencePoint"),
+                                        new XAttribute(XName.Get("vc"), hitCount),
+                                        new XAttribute(XName.Get("uspid"), ++sequencePointMegaCounter),
+                                        new XAttribute(XName.Get("ordinal"), ++sequencePointMiniCounter),
+                                        new XAttribute(XName.Get("sl"), methodPoint.StartLine),
+                                        new XAttribute(XName.Get("sc"), methodPoint.StartColumn),
+                                        new XAttribute(XName.Get("el"), methodPoint.EndLine),
+                                        new XAttribute(XName.Get("ec"), methodPoint.EndColumn)
+                                    );
+                                });
+
+                                var methodElement = new XElement(
+                                    XName.Get("Method"),
+                                    new XAttribute(XName.Get("visited"), allLinesAreVisited),
```

Comment:
Shouldn't we consider the method visited if any sequence point is hit? instead of only if all sequencepoints are hit.

Reply:
Yes, and it has an error too, the lambda is not evaluated until is used, so allLinesAreVisited will be wrong (I'm checking my project output and is wrong)

```
<Method visited="true" isConstructor="false">
        <Name>System.Void SharpReport.RenderImage::RenderPDF(SharpReport.PDF.SimplePDF,SharpReport.PageSize)</Name>
        <FileRef uid="40" />
        <SequencePoint vc="0" uspid="1533" ordinal="1" sl="56" sc="13" el="56" ec="68" />
        <SequencePoint vc="0" uspid="1534" ordinal="2" sl="57" sc="13" el="61" ec="32" />
      </Method>
```

I will fix it and the other suggestions, I'm new at github, I will commit in my branch the fix, it's ok?